types.cc

```diff
@@ -1423,23 +1423,23 @@ struct validate_visitor {
     void operator()(const reversed_type_impl& t) { return t.underlying_type()->validate(underlying, sf); }
     void operator()(const abstract_type&) {}
     template <typename T> void operator()(const integer_type_impl<T>& t) {
-        if (v.empty()) {
+        if (v.size_bytes() == 0) {
```

Comment:
Why not add `empty()` to `FragmentView` instead?

Reply:
API orthogonality was my rationale. Whenever you have both `empty()` and `size_bytes()` available, you have one more decision to make in your life.
But if you think `empty()` is more natural or readable, just say a word and I will switch to it.

Reply:
Most containers have it, even though you can argue it is redundant if they have `size()`. I think it is more natural to read, and you avoid churn in the code too.

Reply:
I thought the same, but what would empty() be? no fragments or size_bytes() == 0?

I guess the latter makes more sense.

Reply:
Is there a difference?

Reply:
Done. Reverted back to `empty()`.